### PR TITLE
FEATURE: Implement oneboxes for chat

### PR DESCRIPTION
**Pull request by nbianca**
This commit implements support for regular and inline oneboxes for chat.
Oneboxes are registered as local onebox using the new plugin API and
are different for links to channels and links to direct messages from
chat.

- [ ] chat-scoped -- core unaffected

### View mode

- [ ] docked (windowed/drawer)
- [ ] isolated (full-screen)

### Browsers

- [ ] safari
- [ ] chrome
- [ ] firefox

### Device

- [ ] desktop – wide screen
- [ ] mobile – `?mobile_view=1`

### Ember

- [ ] ember-cli
- [ ] legacy

### CODE REVIEW

File: lib/onebox/templates/discourse_chat.mustache [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{{^cooked}}
```
Comment (pmusaraj) [inline lib/onebox/templates/discourse_chat.mustache:1]: Is `^` a typo or new syntax? Haven't seen it before.

File: lib/onebox/templates/discourse_chat.mustache [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    <div class="chat-onebox-members">
+      {{#users}}
+        <a class="trigger-user-card" data-user-card="{{username}}" aria-hidden="true" tabindex="-1">
+          <img loading="lazy" alt="{{username}}" width="30" height="30" src="{{avatar_url}}" class="avatar" style="aspect-ratio: 30 / 30;">
+        </a>
+      {{/users}}
+      {{remaining_user_count_str}}
+    </div>
+  </article>
+  <div class="clearfix"></div>
```
Comment (pmusaraj) [inline lib/onebox/templates/discourse_chat.mustache:32]: We still need the clearfix element in oneboxes?

File: lib/onebox/templates/discourse_chat.mustache [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+</aside>
+{{/cooked}}
+
+{{#cooked}}
+<div class="discourse-chat-transcript" data-message-id="{{message_id}}" data-username="{{username}}" data-datetime="{{created_at}}" data-channel-name="{{channel_name}}" data-channel-id="{{channel_id}}">
+  <div class="chat-transcript-user">
+    <div class="chat-transcript-user-avatar">
+      <a class="trigger-user-card" data-user-card="{{username}}" aria-hidden="true" tabindex="-1">
+        <img loading="lazy" alt="{{username}}" width="20" height="20" src="{{avatar_url}}" class="avatar" style="aspect-ratio: 20 / 20;">
```
Comment (pmusaraj) [inline lib/onebox/templates/discourse_chat.mustache:41]: Minor, but the aspect-ratio style is either unnecessary or should be moved to a stylesheet.

**Review by pmusaraj — APPROVED** [review]

File: lib/onebox/templates/discourse_chat.mustache (hunk 1 above)
Comment (CvX) [reply lib/onebox/templates/discourse_chat.mustache:1]: Is the inverted block syntax in mustache 🙃

Comment (pmusaraj) [reply lib/onebox/templates/discourse_chat.mustache:1]: Ohhh... I see, mustache! Ignore me.

Comment (CvX) [reply lib/onebox/templates/discourse_chat.mustache:1]: It's ok, nobody expects mustache 😄 a v-low pri todo is to replace all onebox templates with hbs

File: lib/onebox/templates/discourse_chat.mustache (hunk 2 above)
Comment (jjaffeux) [reply lib/onebox/templates/discourse_chat.mustache:32]: Yes please each time you see clearfix, attempt to remove it please.

**Pull request merged**